exporter/signalfxexporter/factory.go

```diff
@@ -55,13 +55,16 @@ func NewFactory() exporter.Factory {
 
 func createDefaultConfig() component.Config {
 	maxConnCount := defaultMaxConns
+	idleConnTimeout := 30 * time.Second
+
 	return &Config{
 		RetrySettings: exporterhelper.NewDefaultRetrySettings(),
 		QueueSettings: exporterhelper.NewDefaultQueueSettings(),
 		HTTPClientSettings: confighttp.HTTPClientSettings{
```

Comment:
Should we be basing this on an adjusted `confighttp.NewDefaultHTTPClientSettings()` and not building the struct directly?

Reply:
That would be a breaking change, as `NewDefaultHTTPClientSettings` is setting idle_conn_timeout to 90s.
We also set the max number of connections per host to 100, compared to the default of 2.

Reply:
I'm not sure because I remember your stance on avoiding to mutate objects if possible.

Reply:
I think we are good with those changes as they stand, I'd be interested in further guidance for defaults (maybe we can be more opinionated about config defaults down the road).

Reply:
> I'm not sure because I remember your stance on avoiding to mutate objects if possible.

That is applicable to config object passed to the components, not to those that we create :)